libs/git/fileset.go

```diff
@@ -13,10 +13,10 @@ type FileSet struct {
 	view    *View
 }
 
-// NewFileSet returns [FileSet] for the Git repository located at `root`.
-func NewFileSet(root vfs.Path, paths ...[]string) (*FileSet, error) {
+// NewFileSet returns [FileSet] for the directory `root` which is contained within Git repository located at `worktreeRoot`.
```

Comment:
This comment is slightly inaccurate. A Git repository is marked by the presence of a `.git` directory. A Git worktree is marked by the presence of a `.git` file. The `.git` file links back to the location of the original Git direcotory. The worktree root directory can lie inside OR outside the repository root. 

For example:
```
➜  cli git:(DECO-24125--fetch-git-details) git worktree add ../cli-feature
Preparing worktree (new branch 'cli-feature')
HEAD is now at 05ff5086 Properly read git metadata when inside Workspace
➜  cli git:(DECO-24125--fetch-git-details) pwd
/Users/shreyas.goenka/cli
➜  cli git:(DECO-24125--fetch-git-details) cd ../cli-feature
➜  cli-feature git:(cli-feature) cat .git
gitdir: /Users/shreyas.goenka/cli/.git/worktrees/cli-feature
```

Reply:
Makes sense, updated to say "Git worktree" instead of "Git repository".